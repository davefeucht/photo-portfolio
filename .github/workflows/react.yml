# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
