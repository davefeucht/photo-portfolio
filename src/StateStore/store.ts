import { configure, makeAutoObservable } from 'mobx';
import { ImageData, Image, Store, Category, Post, Page } from '../utils/types';

configure({ enforceActions: 'observed' });

class stateStore implements Store {
    menuState = 'closed'
    applicationRoot: HTMLElement = null;
    screenInfo: {
        width: number,
        height: number
    }
    siteInfo: {
        siteName: string,
        siteUrl: string
    }
    visiblePost: {
        postId: number,
        postTitle: string,
        tags: number[],
        tagNames: string[],
        fullImageUrl: string,
        width: number,
        height: number
    };
    categoryList: Category[] = [];
    currentCategoryPosts: Post[] = [];
    currentCategoryData: Category = {
        id: 0,
        count: 0,
        description: '',
        link: '',
        name: '',
        slug: '',
        taxonomy: '',
        parent: 0
    };
    pages: Page[] = [];
    currentPageData: Page = {
        id: 0,
        link: '',
        modified: '',
        slug: '',
        type: '',
        title: {
            rendered: ''
        },
        content: {
            rendered: '',
            protected: false
        },
        author: 0,
        featured_media: 0
    };
    maxItemsPerPage = 10;

    constructor() {
        makeAutoObservable(this);
    }

    setMenuState = (state: string) => {
        this.menuState = state;
    }

    toggleMenuState = () => {
        this.menuState = this.menuState === 'closed' ? 'open' : 'closed';
    }

    setApplicationRoot = (element: HTMLElement) => {
        this.applicationRoot = element;
    }

    setSiteName = (name: string) => {
        this.siteInfo.siteName = name;
    }

    setVisiblePost = (postId: number, postTitle: string) => {
        this.visiblePost.postId = postId;
        this.visiblePost.postTitle = postTitle;
    }

    setVisiblePostImage = (fullImageUrl: string) => {
        this.visiblePost.fullImageUrl = fullImageUrl;
    }

    setVisiblePostTags = (tagNames: string[]) => {
        tagNames.forEach(tagName => {
            this.visiblePost.tagNames.push(tagName);
        });
    }

    clearVisiblePostTagNames = () => {
        this.visiblePost.tagNames.splice(0, this.visiblePost.tagNames.length);
    }

    setCategoryList = (categories: Category[]) => {
        this.categoryList.length = 0;
        categories.forEach((category, index) => {
            this.categoryList[index] = category;
        })
    }

    setCategoryPosts = (posts: Post[]) => {
        this.currentCategoryPosts.length = 0;
        posts.forEach((post, index) => {
            this.currentCategoryPosts[index] = post;
        })
    }

    setCategoryData = (categoryData: Category) => {
        Object.keys(categoryData).forEach((key: keyof Category) => {
            this.currentCategoryData[key] = categoryData[key];
        });
    }

    setThumbnailImageUrl = (imageData: ImageData) => {
        this.currentCategoryPosts[imageData.post_index].thumbnail_image = imageData.image_url;
    }

    setCurrentPost = (postData: Post) => {
        Object.keys(postData).forEach((key: keyof Image) => {
            this.visiblePost[key] = postData[key];
        })
    }

    setPages = (pages: Page[]) => {
        pages.forEach(page => {
            this.pages.push(page);
        });
    }

    setPageData = (pageData: Page) => {
        Object.assign(this.currentPageData, pageData);
    }
}

export default stateStore;
