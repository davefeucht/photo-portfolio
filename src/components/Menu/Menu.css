.menu {
  grid-area: content;
  height: 100%;
  width: 300px;
  background-color: black;
  transition: opacity 0.5s ease-in-out;
  z-index: 10;

  color: #FFFFFF;
  font-size: 22px;
}

.menu.open {
  opacity: 0.9;
}
.menu.closed {
  opacity: 0;
  pointer-events: none;
}

.menu ul {
    list-style: none;
}