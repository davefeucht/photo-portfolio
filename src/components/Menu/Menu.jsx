/** ***************
* Menu component implements the menu of the application
**************** */

import './Menu.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';

import MenuLink from '../MenuLink/MenuLink.jsx';

const Menu = ({ stateStore }) => {
    const {
        menuState, toggleMenuState, categoryList, pages
    } = stateStore;
    return (
        <div className={`menu ${menuState}`}>
            <ul>
                <li onClick={() => toggleMenuState()}>
                    <MenuLink href="" text="Home" />
                </li>
                <li>
                    Categories
                    <ul>
                        {categoryList.map(category => {
                            return (
                                <li key={`li_${category.name}_${category.id}`} onClick={() => toggleMenuState()}><MenuLink key={`${category.name}_${category.id}`} href={`category/${category.id}`} text={category.name} /></li>
                            );
                        })}
                    </ul>
                </li>
                <li>
                    Info
                    <ul>
                        {pages.map(page => {
                            return (
                                <li key={`li_${page.title.rendered}`} onClick={() => toggleMenuState()}><MenuLink key={`${page.title.rendered}_${page.id}`} href={`page/${page.id}`} text={page.title.rendered} /></li>
                            );
                        })}
                    </ul>
                </li>
            </ul>
        </div>
    );
};

Menu.displayName = 'Menu';

Menu.propTypes = {
    stateStore: PropTypes.object.isRequired
};

export default observer(Menu);
