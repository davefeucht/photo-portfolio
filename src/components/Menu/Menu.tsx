/** ***************
* Menu component implements the menu of the application
**************** */

import './Menu.css';

import { observer } from 'mobx-react';
import * as React from 'react';

import { Store } from '../../utils/types';
import MenuLink from '../MenuLink/MenuLink';

interface MenuProps {
    stateStore: Store
}

const Menu: React.FC<MenuProps> = ({ stateStore }) => {
    const {
        menuState, toggleMenuState, categoryList, pages
    } = stateStore;
    return (
        <div className={`menu ${menuState}`}>
            <ul>
                <li onClick={() => toggleMenuState()}>
                    <MenuLink href="" text="Home" />
                </li>
                <li>
                    Categories
                    <ul>
                        {categoryList.map(category => {
                            return (
                                <li key={`li_${category.name}_${category.id}`} onClick={() => toggleMenuState()}><MenuLink key={`${category.name}_${category.id}`} href={`category/${category.id}`} text={category.name} /></li>
                            );
                        })}
                    </ul>
                </li>
                <li>
                    Info
                    <ul>
                        {pages.map(page => {
                            return (
                                <li key={`li_${page.title.rendered}`} onClick={() => toggleMenuState()}><MenuLink key={`${page.title.rendered}_${page.id}`} href={`page/${page.id}`} text={page.title.rendered} /></li>
                            );
                        })}
                    </ul>
                </li>
            </ul>
        </div>
    );
};

Menu.displayName = 'Menu';

export default observer(Menu);
