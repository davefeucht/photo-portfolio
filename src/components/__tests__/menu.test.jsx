import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import StateStore from '../../StateStore/store';
import Menu from '../Menu/Menu';

test('Menu displays', async () => {
    const store = new StateStore();
    const { container } = render(
        <MemoryRouter>
            <Menu stateStore={store} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
