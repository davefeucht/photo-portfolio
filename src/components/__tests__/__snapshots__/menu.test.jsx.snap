// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu displays 1`] = `
<div
  class="menu closed"
>
  <ul>
    <li>
      <a
        href="/"
      >
        <div
          class="menu-link"
        >
          Home
        </div>
      </a>
    </li>
    <li>
      Categories
      <ul />
    </li>
    <li>
      Info
      <ul />
    </li>
  </ul>
</div>
`;
